Source/player.cpp

```diff
@@ -484,9 +484,27 @@ void NewPlrAnim(int pnum, BYTE *Peq, int numFrames, int Delay, int width, int nu
 	plr[pnum]._pAnimDelay = Delay;
 	plr[pnum]._pAnimWidth = width;
 	plr[pnum]._pAnimWidth2 = (width - 64) >> 1;
-	plr[pnum]._pAnimNumSkippedFrames = numSkippedFrames;
 	plr[pnum]._pAnimGameTicksSinceSequenceStarted = processAnimationPending ? -1 : 0;
-	plr[pnum]._pAnimStopDistributingAfterFrame = stopDistributingAfterFrame;
+	plr[pnum]._pAnimRelevantAnimationFramesForDistributen = 0;
+	plr[pnum]._pAnimGameTickModifier = 0.0f;

```

Comment:
I believe this is the planned next step in terms of extending this to also work for monsters etc.
(At the very least diablo in Hellfire has a sped up animation)

Reply:
As mentioned in #1468 this would be the next step after this pr.
I try to do small pr's so they are easier to review.

Reply:
> As mentioned in #1468 this would be the next step after this pr.

You should consider reverting the order. Doing cleanup PRs first and then making a change results in simpler code than doing it in the opposite order.

Reply:
In this case the logic wouldn't be simpler. It would be simple shifted to another class.
okay this also reduces the complexity a little. :grin:

but you should consider I didn't plan out to everything at the beginning.
And I think that's normal in agile software development.
As everything else the Animation Distrubtion Logic advances step by step.

But I will try to do the refactoring first if it's foreseeable how the layout of new code will be or what will be needed. 🙂